packages/interactivity/src/store.ts

```diff
@@ -207,6 +207,24 @@ interface StoreOptions {
 	lock?: boolean | string;
 }
 
+// Utility type to check if a function is a generator function.
+type IsGenerator< T > = T extends (
+	...args: infer A
+) => Generator< infer R, any, any >
+	? ( ...args: A ) => Promise< R >
+	: never;
```

Comment:
This doesn't look quite right, the generators should _yield_ promises, but I think the async function this transforms to should resolve to the returned value. Probably something more like this:

```suggestion
) => Generator< any, infer R, any >
	? ( ...args: A ) => Promise< R >
	: never;
```

The first type argument could be `Promise< any >`, but that's probablye too strict. We await the yielded values and you can await values of any type (although it only makes sense for promises).

The name `IsGenerator` is a bit strange because this actually does the conversaion.

Reply:
Indeed, thank you 🙂

Fixed in https://github.com/WordPress/gutenberg/pull/62400/commits/5c73a529452d0b8c82ec0dc1827681e5a198ead2.